samples/helloworld_esm/worker.js

```diff
@@ -1,5 +1,26 @@
+let workerCount = 0;
+
 export default {
-  async fetch(req, env) {
-    return new Response("Hello World\n");
+  async fetch(_req, env) {
+    const esModule = `
+    console.log('hello from inner js')
+    export default {
+      async task() {
+        return 'this is a CID';
+      }
+    }
+    `
+    const workerConfig = JSON.stringify({
+      services: [{
+        name: `inner-main-${++workerCount}`,
+        worker: {
+          modules: [
+            {name: 'worker', esModule },
+          ],
+          compatibilityDate: '2023-02-28',
+        }
+      }],
+    });
+    return new Response(`${await env.workerd.runWorker(workerConfig)}\n`);
```

Comment:
If we were to introduce this ability, either hiding it behind `fetch` or having some API that is as close as possible to the generally standard `new Worker()` syntax would be ideal. I'm particularly concerned about embedded the config format into the worker syntax like this. My preference would be something like...

```js
await env.otherWorker.fetch(...)
```

or 

```
const otherWorker = new Worker('specifier')
otherWorker.postMessage(...)
otherWorker.addEventListener('message', () => {});
```

Both of which use existing, idiomatic API patterns that are well established across multiple runtimes as opposed to inventing new API.

Reply:
Note that the model presented here also goes against our policy of not allowing dynamic runtime evaluation (e.g. new `evail('...')` or `new Function('...')` type calls) since the `workerConfig` with the content of the `esModule` is constructed at runtime.

Reply:
Thanks for the (very quick!) review. This is very early work that I cobbled together in a couple of days, so I appreciate your providing early direction.

I like the established `Worker` API. I will read through that spec and see if there's anything that lends itself to pluggable specifiers such that dynamic runtime evaluation is optional.

I more generally worry that this is not the canonical way to implement the worker hypervisor, and that the correct use of the tool is to put it outside. It just seemed very natural to have a self-capability?

Reply:
I'm trying to implement [the diagram](https://developers.cloudflare.com/workers/learning/security-model/). It appeared as though it was intended for everything to be run as services, including the "scheduling and routing" box :)

![image](https://github.com/cloudflare/workerd/assets/2353785/50a42611-a5f3-4f77-86ed-f8b1132e6022)